common/remote/sockfile.cpp

```diff
@@ -2598,8 +2598,7 @@ void CRemoteFile::copyTo(IFile *dest, size32_t buffersize, ICopyFileProgress *pr
 
 unsigned getRemoteVersion(ISocket * socket, StringBuffer &ver)
 {
-    static CriticalSection sect;
-    CriticalBlock block(sect);
+    // used to have a global critical section here
```

Comment:
Odd, I cannot see why it was there either.

You might want to prevent multiple threads talking to the *same socket* , but that isn't done elsewhere either.

So looks like a good change that will help applyNodes and also CSashaDaFSMonitorServer::checkCluster